aptos-move/aptos-vm/src/delta_state_view.rs

```diff
@@ -32,9 +33,14 @@ where
     }
 
     fn get_state_value(&self, state_key: &StateKey) -> Result<Option<StateValue>> {
-        match self.write_set.get(state_key) {
-            Some(write_op) => Ok(write_op.as_state_value()),
-            None => self.base.get_state_value(state_key),
+        match self.change_set.delta_change_set().get(state_key) {
```

Comment:
One question here is whether we want to adjust the write_set and delta_set when we resolve something from the delta_set. Maybe it's not that relevant? @georgemitenkov

Reply:
Do you mean re-cache materialized delta as a write op?

Reply:
yes, but thinking about it, shouldn't really matter here